### PR TITLE
Stripe: Handle upstream processing errors

### PR DESCRIPTION
**Pull request by bpollack**
Sometimes, Stripe can fail internally, in which case its status will be set to "failed" even though an error key may be absent. This commit handles that case and adds a regression test for it.

Unit: 127 tests, 674 assertions, 0 failures, 0 errors, 0 pendings, 0 omissions, 0 notifications
100% passed

Remote: 64 tests, 288 assertions, 0 failures, 0 errors, 0 pendings, 0 omissions, 0 notifications
100% passed

### CODE REVIEW

**Comment by bpollack** [comment]
Just updated after a clean run on remote tests (Stripe didn't time out, whee).

File: lib/active_merchant/billing/gateways/stripe.rb [hunk 1]
```diff
@@ -549,14 +549,15 @@ def commit(method, url, parameters = nil, options = {})
         add_expand_parameters(parameters, options) if parameters
         response = api_request(method, url, parameters, options)
 
-        success = !response.key?("error")
+        success = !response.key?("error") && response["status"] != "failed"
+        error_message = response.fetch("error", {"message" => "No details provided"})["message"]
```
Comment (curiousepic) [inline lib/active_merchant/billing/gateways/stripe.rb]: "No error details" might be a little more explicit than "No details provided".

Comment (bpollack) [reply lib/active_merchant/billing/gateways/stripe.rb]: Good call; will change.

**Comment by bpollack** [comment]
All right, give that a shot @curiousepic ; it makes the patch larger, but I think I agree it makes it easier to read.

**Comment by curiousepic** [comment]
@bpollack I'll take readable over compact any day. Honestly, this was more about `.fetch` being new/unintuitive for me and giving the logic room to breath past one line, but not a big deal, I do like it and might start using it more!

**Comment by bpollack** [comment]
@curiousepic `Hash#dig` negates 99% of the need I've got for `Hash#fetch`, so I wouldn't get too attached for it if we can indeed drop Ruby 2.1 and 2.2 support for AM shortly. :)

**Pull request closed without merging**